lib/hawkular/metrics/metric_api.rb

```diff
@@ -129,6 +129,19 @@ def get_data(id, starts: nil, ends: nil, bucketDuration: nil, buckets: nil, perc
         resp.is_a?(Array) ? resp : [] # API returns no content (empty Hash) instead of empty array
       end
 
+      # Retrieve raw data for multiple metrics.
+      # This API is experimental and may undergo non-backwards compatible changes on the future
+      # @param ids [Array[String]] metric definition ids
+      # @param starts [Integer] optional timestamp (default now - 8h)
+      # @param ends [Integer] optional timestamp (default now)
+      # @param limit [Integer] optional limit the number of data points returned
+      # @param order [String] optional Data point sort order, based on timestamp (ASC, DESC)
+      # @return [Array[Hash]] named datapoints
+      def raw_data(ids, starts: nil, ends: nil, limit: nil, order: nil)
+        params = { ids: ids, start: starts, end: ends, limit: limit, order: order }
+        @client.http_post("/#{@resource}/raw/query", params)
```

Comment:
Believe me, you don't want to know that :-)

Seriously: metric ids can be very long (strings with crazy characters) and querying several of them at once can lead to very long and complicated query parts of the GET requests, so the hawkular-metrics team decided to deliver the metric ids as payload of a POST request.
I don't think it is that bad, because the caching argument (GET vs POST) is not really applicable for metric data, as different requests usually ask for different time ranges and return different data each time.